flow.py

```diff
@@ -1,37 +1,46 @@
 import sys
+from packet import Packet
+from datapacket import DataPacket
+from acknowledgementpacket import AcknowledgementPacket
+from routingpacket import RoutingPacket
+from constants import *
 
 class Flow(object):
     """A flow class that represents active connections between
     hosts and routers.
 
     Attributes:
         ns (NetworkSimulator): Instance of the NetworkSimulator class
-        flow_id (int): Unique id identifying the flow
-        src (Node): The flow's source node
-        dest (Node): The flow's destination node
-        data_amount (int): Data capacity of the flow
+        flow_id (string): Unique id identifying the flow
+        src (Node): The flow's source node id
+        dest (Node): The flow's destination node id
+        data_amount (float): Data capacity of the flow (bits)
 
-        window_size (int): The size of the window
+        start_time (float): Start time in seconds
         unacknowledged_packets (dict): The list of packets with no acknowledgement
-        current_packet (Packet): The current packet in the flow
-        num_packets (int):  The number of packets that has been sent so far
-        last_acknowledged (Packet): The last acknowledged packet in the flow
+        timed_out_packets (dict): The list of packets timed out before acknowledgement
+        num_packets_sent(float): Number of packets that have been sent through
+            the flow
+        num_packets (float):  Number of packets to be sent through the flow
+        window_size (float): The size of the window
 
     """
 
-    def __init__(self, ns, flow_id, src, dest, data_amount):
+    def __init__(self, ns, flow_id, src, dest, data_amount, start_time,
+        window_size=1):
         self.ns = ns
         self._flow_id = flow_id
         self._src = src
         self._dest = dest
         self.data_amount = data_amount
+        self.start_time = start_time
+        self.num_packets_sent = 0
+        self.unacknowledged_packets = set()
+        self.timed_out_packets = set()
+        self.num_packets = data_amount/DATA_PACKET_SIZE
```

Comment:
is data amount an int/ do you need to round this? Either way, don't you need an extra packet to send any left over data 
that might be smaller than data_packet_size just in case the division leaves a remainder?